### PR TITLE
Use our own ConsoleOutput class over SymfonyStyle

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

Extracts console output logic from the `InfectionCommand` class into its own class.

### CODE REVIEW

File: src/Console/ConsoleOutput.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        throw MsiCalculationException::create('min-msi');
+    }
+
+    public function logBadCoveredMsiErrorMessage(MetricsCalculator $metricsCalculator, float $minCoveredMsi)
+    {
+        if ($minCoveredMsi) {
+            $this->io->error(
+                    sprintf(
```
Comment (borNfreee) [inline src/Console/ConsoleOutput.php]: incorrect indentation

File: src/Console/ConsoleOutput.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        $this->io->note('Running infection with an unknown log-verbosity option, falling back to ' . $default . ' option');
+    }
+
+    public function logInitialTestsDoNotPass(Process $initialTestSuitProcess, string $testFrameworkKey)
+    {
+        $lines = [
+            'Project tests must be in a passing state before running Infection.',
+            sprintf(
+                '%s reported an exit code of %d.',
+                ucfirst($testFrameworkKey),
```
Comment (borNfreee) [inline src/Console/ConsoleOutput.php]: while we at it, could you please change it to `AbstractTestFrameworkAdapter::getName()`

File: src/Command/InfectionCommand.php [hunk 3]
```diff
@@ -511,8 +427,53 @@ protected function initialize(InputInterface $input, OutputInterface $output)
...
     }
+
+    /**
+     * Checks if the infection run passed the constraints like min msi & covered msi.
+     *
+     * @param MetricsCalculator $metricsCalculator
+     *
+     * @return int
+     */
+    private function didRunPassConstraints(MetricsCalculator $metricsCalculator): int
```
Comment (sidz) [inline src/Command/InfectionCommand.php]: should it be smth like `hasRunPassConstraints` or `isRunPassConstraints` ?

File: src/Console/ConsoleOutput.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            $this->io->error(
+                sprintf(
+                    self::CI_FLAG_ERROR,
+                    'MSI',
+                    $minMsi,
+                    $metricsCalculator->getMutationScoreIndicator()
+                )
+            );
+
+            return;
```
Comment (sidz) [inline src/Console/ConsoleOutput.php]: maybe it'd be better to do smth like:
```
if (!$minMsi) {
    throw new exception...
}

$this->io->error...
```
It'll remove useless `return`

File: src/Console/ConsoleOutput.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+        throw MsiCalculationException::create('min-msi');
+    }
+
+    public function logBadCoveredMsiErrorMessage(MetricsCalculator $metricsCalculator, float $minCoveredMsi)
+    {
+        if ($minCoveredMsi) {
```
Comment (sidz) [inline src/Console/ConsoleOutput.php]: the same here

File: src/Command/InfectionCommand.php (hunk 3 above)
Comment (BackEndTea) [reply src/Command/InfectionCommand.php]: How about `hasRunPassedConstraints` ?

**Comment by sanmai** [comment]
I guess now we can have a test suite for `ConsoleOutput`. Do you think this should go to another PR?

**Comment by borNfreee** [comment]
Thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @borNfreee @sanmai

**Pull request merged**